src/prediction.jl

```diff
@@ -1,37 +1,79 @@
-non_na_mean(vv) = mean(vv[.!(isnan.(vv))])
+########################################
+# 0. Hierarchical Structures and Types #
+########################################
 
-function fix_metadata_colnames!(longdata_df::DataFrame)
+abstract type Prediction end
+struct Classification <: Prediction end
+struct Regression <: Prediction end
 
-    ## This function was created to fix small inconsistencies and typos on the longdata table.
-    ## As original data cleanup and wrangling progresses, it will be altered and ultimately deprecated.
-    ## Comments are provided to help review transformations
+abstract type Predictor end
+abstract type UnivariatePredictor <: Resonance.Predictor end
+abstract type MultivariatePredictor <: Resonance.Predictor end
 
-    # 1. Rename "[Collinsella]_massiliensis" to "Collinsella_massiliensis"
-    longdata_df[longdata_df.variable .== "[Collinsella]_massiliensis", :variable] .= "Collinsella_massiliensis"
+mutable struct UnivariateRandomForestClassifier <: Resonance.UnivariatePredictor
+    name::String
+    original_data::DataFrame
+    inputs_outputs::Tuple{DataFrame, CategoricalArrays.CategoricalVector{Bool, UInt32, Bool, CategoricalArrays.CategoricalValue{Bool, UInt32}, Union{}}}
```

Comment:
Is there a reason that this is a single parameter instead of having `.inputs` and `.outputs`?

Reply:
I also wonder if it's necessary / desirable to give such specific type constraints...

Reply:
1 - There's not a powerful reason, no. It was my first instinct, and I implemnted it as such. It's probably because at some point I was writing the line `X, y = foo[:bar'` and made it a tuple then? But yeah, if you think its better to have inputs and outputa separated, let's do this.

2 - This is a verbatim copy of `typeof(bar)` where `bar = coerce(classifier_fun(y), OrderedFactor)